deltadb/src/deltadb_expr.c

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright (C) 2015- The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file COPYING for details.
+*/
+
+#include "jx.h"
+
+#include <string.h>
+#include <stdio.h>
+
+struct deltadb_expr {
+	char operator[32];
+	char *param;
+	char *val;
+	struct deltadb_expr *next;
+};
+
+struct deltadb_expr * deltadb_expr_create( const char *str, struct deltadb_expr *next )
+{
+	struct deltadb_expr *e = malloc(sizeof(*e));
+	e->param = strdup(str);
+	char *delim = strpbrk(e->param, "<>=!");
+	e->val = strpbrk(delim, "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ");
+	int operator_size = (int)(e->val-delim);
+	strncpy(e->operator,delim,operator_size);
```

Comment:
This is going to segfault if e->param does not contain an operator. (delim and e->val could be NULL).